usecases/modules/compare.go

```diff
@@ -33,6 +33,9 @@ func reVectorize(ctx context.Context,
 	targetVector string,
 	findObjectFn modulecapabilities.FindObjectFn,
 ) (bool, models.AdditionalProperties, []float32, error) {
+	if ctx.Err() != nil {
+		return false, nil, nil, fmt.Errorf("context is done: %w", ctx.Err())
+	}
```

Comment:
```suggestion
	if ctx.Err() != nil {
		return false, nil, nil, ctx.Err()
	}
```
Does `"context is done: "` part of the message adds any valuable information?
`Context.Err` will contain the reason: "canceled" or "deadline exceeded".

Reply:
you also check the context in the loop calling this function, resulting in two context checks per loop iteration